nifi-nar-bundles/nifi-irc-bundle/nifi-irc-client-service/src/main/java/org/apache/nifi/irc/StandardIRCClientService.java

```diff
@@ -0,0 +1,233 @@
+                .nick(context.getProperty(IRC_NICK).getValue())
+                .user("nifi")
+                .realName(String.join(" - ", new String[] { this.getClass().getSimpleName(), this.clientIdentification }));
+
+        // Setup Security
+        final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+
+        if (sslContextService == null) {
+            // Disabled...
+            clientSkeleton.secure(false);
+        } else {
+            // Enabled
+            clientSkeleton.secure(true);
+            SSLContext sslContext;
+            TrustManagerFactory tmf;
+
+            // Is key configured? If yes, populate and let it go...
+            if (sslContextService.isKeyStoreConfigured()) {
+                final String keyPassword = sslContextService.getKeyPassword();
+                final String keyFile = sslContextService.getKeyStoreFile();
+
+                if (!StringUtils.isEmpty(keyPassword) && !StringUtils.isEmpty(keyFile) ) {
+                    clientSkeleton.secureKeyPassword(keyPassword);
+                    clientSkeleton.secureKey(new File(keyFile));
+                    sslContext = sslContextService.createSSLContext(SSLContextService.ClientAuth.REQUIRED);
+                } else {
+                    sslContext = sslContextService.createSSLContext(SSLContextService.ClientAuth.NONE);
+                }
+
+                try {
+                    Provider tlsProvider = sslContext.getProvider();
+
+                    final KeyStore trustStore = KeyStoreUtils.getTrustStore(sslContextService.getTrustStoreType());
+                    try (final InputStream trustStoreStream = new FileInputStream(new File(sslContextService.getTrustStoreFile()))) {
+                        trustStore.load(trustStoreStream, sslContextService.getTrustStorePassword().toCharArray());
+                    }
+
+                    tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                    tmf.init(trustStore);
+
+                    clientSkeleton.secureTrustManagerFactory(tmf);
+                } catch (NoSuchAlgorithmException | CertificateException | KeyStoreException | IOException e) {
+                    logger.error("Failed to initialize secure IRC service due to {}", new Object[]{e.getMessage()}, e);
+                }
+
+            }
+        }
+        this.ircClient = clientSkeleton.build();
+
+        // Setup the Server Handlers
+        ServiceEventHandler serviceEventHandler = new ServiceEventHandler(connectionStatus, requestedChannels, getLogger());
+        ircClient.getEventManager().registerEventListener(serviceEventHandler);
+    }
+
+    @OnDisabled
+    public void shutdown() {
+        this.ircClient.shutdown(clientIdentification + " - is going to rest a bit...");
+        while (this.connectionStatus.get()) {
+            // Wait for the disconnection
```

Comment:
probably should add some time based kick out here or else we could be waiting a longgg time